harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -595,12 +598,16 @@ def _compute_validation_metrics(self) -> workload.Response:
                 )
                 callback.on_validation_step_end(metrics)
 
+        step_duration = time.time() - step_start_time
```

Comment:
i'm not sure whether it's better to have duration end here or after the `on_validation_end` callbacks.  i'm leaning towards after the callbacks since that is technically when the validation workload completes.  what do you think?

Reply:
Made the change to include the pre- and post-workload callbacks in the timings.  Also fixed another bug or two.  Please re-review.